build.js

```diff
@@ -51,36 +67,41 @@ function makeComponentDictionary(globPattern) {
 
 	const componentSourceFiles = glob.sync(globPattern)
 	const components = {}
+	// loop over found components matching the glob pattern
 	for (const componentSourceFile of componentSourceFiles) {
 		const componentName = componentSourceFile
 			.split('/').pop() // get the actual filename
 			.replace(/\.html$/, '') // remove .html at the end
 	
 		console.log('Found component', componentName)
 
-		// parse component
+		// parse component (separate into subcomponents if it is compound)
 		const fileString = fs.readFileSync(componentSourceFile, "utf8")
-		if (fileString.indexOf('TEMPORARY_OPENING_SIGNATURE ') == 0) {
+		if (fileString.indexOf(OPENING_SIGNATURE) == 0) {
 			console.log(componentName, 'was identified as a compound component')
 
-			// split by sections
-			let sections = fileString.split('\r\nTEMPORARY_OPENING_SIGNATURE ')
-			sections[0] = sections[0].replace('TEMPORARY_OPENING_SIGNATURE ', '')
-			// seperate sections into name and content
-			sections = sections.map(section => section.split(' TEMPORARY_CLOSING_SIGNATURE\r\n'))
+			// split by sections / subcomponents
+			let sections = fileString.split('\r\n' + OPENING_SIGNATURE)
+			sections[0] = sections[0].replace(OPENING_SIGNATURE, '')
+			// separate sections into name and content
+			sections = sections.map(section => section.split(CLOSING_SIGNATURE + '\r\n'))
 			// add component sections to dictionary
 			let componentDict = {}
-			sections.map(section => componentDict[section[0]] = new Handlebars.SafeString(section[1]))
+			sections.forEach(
+				section => 
+				componentDict[section[0]] = new Handlebars.SafeString(section[1])
+				)
```

Comment:
in general, it's nicer to have the same indentation for the line with closing bracket as for the line with opening bracket. (ie if this had one tab less, it would be nicer)